qurro/qarcoal.py

```diff
@@ -31,21 +31,32 @@ def filter_and_join_taxonomy(feat_table, taxonomy, num_string, denom_string):
         denom_df: pd.DataFrame of denominator features x samples
     """
 
+    feat_table_copy = feat_table.copy()
+
+    # drop samples with total 0 features
+    # this is done here as well as later on based on samples w/o
+    # matching features because of some problems with filtering
+    # introducing NAs
+    feat_table_copy = feat_table_copy.loc[
+        :, (feat_table_copy != 0).any(axis=0)
+    ]
```

Comment:
I think this could be written more concisely as 
```python
    feat_table_copy = feat_table.loc[
        :, (feat_table != 0).any(axis=0)
    ]
```